craco.config.cjs

```diff
@@ -29,25 +32,30 @@ module.exports = {
     },
   },
   webpack: {
-    plugins: [
-      new VanillaExtractPlugin({ identifiers: 'short' }),
-      new DefinePlugin({
-        'process.env.REACT_APP_GIT_COMMIT_HASH': JSON.stringify(commitHash.toString()),
-      }),
-    ],
+    plugins: [new VanillaExtractPlugin({ identifiers: 'short' })],
     configure: (webpackConfig) => {
-      const instanceOfMiniCssExtractPlugin = webpackConfig.plugins.find(
-        (plugin) => plugin instanceof MiniCssExtractPlugin
-      )
-      if (instanceOfMiniCssExtractPlugin !== undefined) instanceOfMiniCssExtractPlugin.options.ignoreOrder = true
+      // Extend process.env with dynamic values (eg commit hash).
+      Object.assign(webpackConfig.plugins.find((plugin) => plugin instanceof DefinePlugin).definitions['process.env'], {
+        REACT_APP_GIT_COMMIT_HASH: JSON.stringify(commitHash),
+      })
+
+      if (isProduction) {
+        // Type checking and linting are only necessary as part of development and testing.
+        // Omit them from production builds, as they slow down the feedback loop.
+        webpackConfig.plugins = webpackConfig.plugins.filter((plugin) => {
```

Comment:
is there a way we can do this via some config when we upgrade CRA or move to next? feels a bit weird to have to search through plugins to modify

Reply:
No, this is the documented way to do so: the only way to do this, AFAIK, is to eject or use a package like craco that lets you modify it through a function exactly like this.